lib/Fmt_ast.ml

```diff
@@ -2799,78 +2787,84 @@ and fmt_cases c ctx cs =
         | None -> Stop None)
       ~finish:(fun acc -> Some acc)
   in
-  let max_len_name = fold_pattern_len ~f:max cs in
-  list_fl cs (fun ~first ~last ({pc_lhs; pc_guard; pc_rhs} as case) ->
-      let xrhs = sub_exp ~ctx pc_rhs in
-      let indent =
-        match
-          (c.conf.cases_matching_exp_indent, (ctx, pc_rhs.pexp_desc))
-        with
-        | ( `Compact
-          , ( Exp {pexp_desc= Pexp_function _ | Pexp_match _ | Pexp_try _; _}
-            , (Pexp_match _ | Pexp_try _) ) ) ->
-            2
-        | _, _ -> c.conf.cases_exp_indent
-      in
-      let align_nested_match =
-        match (pc_rhs.pexp_desc, c.conf.nested_match) with
-        | (Pexp_match _ | Pexp_try _), `Align -> last
-        | _ -> false
-      in
-      let parens_here, parens_for_exp =
-        if align_nested_match then (false, Some false)
-        else if c.conf.leading_nested_match_parens then (false, None)
-        else (parenze_exp xrhs, Some false)
-      in
-      (* side effects of Cmts.fmt_before before [fmt_lhs] is important *)
-      let leading_cmt = Cmts.fmt_before c pc_lhs.ppat_loc in
-      let xlhs = sub_pat ~ctx pc_lhs in
-      let paren_lhs =
-        match pc_lhs.ppat_desc with
-        | Ppat_or _ when Option.is_some pc_guard -> true
-        | _ -> parenze_pat xlhs
-      in
-      let eol =
-        Option.some_if
-          (Cmts.has_before c.cmts pc_rhs.pexp_loc)
-          (fmt "@;<1000 0>")
-      in
-      let indent = if align_nested_match then 0 else indent in
-      let fmt_padding =
-        let level = match c.conf.break_cases with `Nested -> 2 | _ -> 3 in
-        fmt_if_k
-          ( c.conf.align_cases
-          && not (Cmts.has_after c.cmts xlhs.ast.ppat_loc) )
-          ( match (max_len_name, pattern_len case) with
-          | Some max_len, Some len ->
-              let pad = String.make (max_len - len) ' ' in
-              fmt_or_k
-                Poly.(c.conf.break_cases = `All)
-                (str pad)
-                (fits_breaks ~level "" pad)
-          | _ -> noop )
+  let max_len = fold_pattern_len ~f:max cs in
+  let level = match c.conf.break_cases with `Nested -> 2 | _ -> 3 in
+  list_fl cs (fun ~first ~last case ->
+      let padding =
+        let xlhs = sub_pat ~ctx case.pc_lhs in
+        let add_padding =
+          c.conf.align_cases && not (Cmts.has_after c.cmts xlhs.ast.ppat_loc)
+        in
+        match max_len with
+        | Some max_len when add_padding -> (
+          match pattern_len case with
+          | Some pattern_len ->
+              let pad = String.make (max_len - pattern_len) ' ' in
+              Some
+                (fmt_or_k
+                   Poly.(c.conf.break_cases = `All)
+                   (str pad)
+                   (fits_breaks ~level "" pad))
```

Comment:
I agree the binds were looking better than nested matches. That's not important.